webpack/assets/javascripts/react_app/components/ConfirmModal/ConfirmModal.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+import { Modal, Button, ModalVariant } from '@patternfly/react-core';
+import { translate as __ } from '../../common/I18n';
+import { noop } from '../../common/helpers';
+import { closeConfirmModal } from './ConfirmModalActions';
+import {
+  selectConfirmCB,
+  selectIsOpen,
+  selectLabel,
+  selectMessage,
+  selectModalProps,
+} from './ConfirmModalSelectors';
+
+export const ConfirmModal = props => {
+  const [isLoading, setLoading] = React.useState(false);
+  const isOpen = useSelector(selectIsOpen);
+  const label = useSelector(selectLabel);
+  const message = useSelector(selectMessage);
+  const onConfirm = useSelector(selectConfirmCB);
+  const modalProps = useSelector(selectModalProps);
+
+  const dispatch = useDispatch();
+  const closeModal = () => dispatch(closeConfirmModal());
+  const handleConfirm = () => {
+    setLoading(true);
+    onConfirm(closeModal);
+  };
+
+  return (
+    <Modal
+      id="app-confirm-modal"
+      variant={ModalVariant.small}
+      title={label}
+      isOpen={isOpen}
+      onClose={closeModal}
+      actions={[
+        <Button
+          key="confirm"
+          variant="primary"
+          onClick={handleConfirm}
+          isLoading={isLoading}
+          isDisabled={isLoading}
+        >
+          {__('Confirm')}
+        </Button>,
+        <Button key="cancel" variant="link" onClick={closeModal}>
+          {__('Cancel')}
+        </Button>,
+      ]}
+      {...modalProps}
+    >
+      {message}
+    </Modal>
+  );
+};
+
+ConfirmModal.propTypes = {
+  label: PropTypes.string,
+  message: PropTypes.string,
```

Comment:
I think message should be a string or a react component in case we want to use links in the modal for example